.evergreen/run-checks.sh

```diff
@@ -3,6 +3,14 @@ set -o errexit # Exit the script with error if any of the commands fail
 
 source "${PROJECT_DIRECTORY}/.evergreen/init-nvm.sh"
 
+set -o xtrace
+
+## Attempt to update our EVG config
+# if it changes, crash so that any gen script changes are forced to be run before pushing
+npm run build:evergreen
+git diff --exit-code ./.evergreen/config.yml
```

Comment:
do you have a patch that shows this working? What does the error look like? Will it be clear from the logs what failed and why?

Reply:
[here](https://evergreen.mongodb.com/task_log_raw/mongo_node_driver_lint_run_checks_patch_79f307cb2858e527e97681638db6251eaaa8e227_616750e80305b97de980d196_21_10_13_21_34_38/0?type=T#L159)

Reply:
can we add a log ahead of the task execution maybe that clearly states its purpose so that it's clear what to do about the failure to someone who is seeing it for the first time?

Reply:
certainly, it is done, a little verbose with the set xtrace on, but it looks like this:
```
+ echo 'Running evergreen config generation, expecting no changes...'
Running evergreen config generation, expecting no changes...
+ npm run build:evergreen

> mongodb@4.1.3 build:evergreen
> node .evergreen/generate_evergreen_tasks.js

+ git diff --exit-code ./.evergreen/config.yml
+ echo 'Evergreen unexpectedly changed!'
Evergreen unexpectedly changed!
+ echo 'Did you run: node .evergreen/generate_evergreen_tasks.js'
Did you run: node .evergreen/generate_evergreen_tasks.js
```